src/typing_extensions.py

```diff
@@ -4222,6 +4224,44 @@ def evaluate_forward_ref(
         )
 
 
+class Sentinel:
+    """Create a unique sentinel object.
+
+    *name* should be the fully-qualified name of the variable to which the
```

Comment:
Fully-qualified? That doesn't match the PEP which does `MISSING = Sentinel("MISSING")`.

Reply:
This was copied verbatim from https://github.com/taleinat/python-stdlib-sentinels/, updated.

Reply:
Actually the repr construction by splitting dots don't make sense anymore, so I also updated the logic in the last logic.

Reply:
It'd make sense to be fully qualified inside the module, so `Sentinel("SomeClass.CONST")`. Pickle expects that if we decide to use it. It's recommended in the PEP under "Additional Notes", but not required.

Reply:
> It'd make sense to be fully qualified inside the module, so `Sentinel("SomeClass.CONST")`. Pickle expects that if we decide to use it. It's recommended in the PEP under "Additional Notes", but not required.

This will have to be decided in the PEP discussion.